plugins/module_utils/email_alert.py

```diff
@@ -0,0 +1,176 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2023, XLAB Steampunk <steampunk@xlab.si>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+from __future__ import annotations
+
+__metaclass__ = type
+
+from .rest_client import RestClient
+
+# from .client import Client
+from ..module_utils.utils import PayloadMapper, get_query
+
+from ..module_utils.typed_classes import (
+    TypedTaskTag,
+    TypedEmailAlertToAnsible,
+    TypedEmailAlertFromAnsible,
+)
+from typing import Union, Any, Dict
+
+
+class EmailAlert(PayloadMapper):
+    def __init__(
+        self,
+        uuid: Union[str, None] = None,
+        alert_tag_uuid: Union[str, None] = None,
+        email_address: Union[str, None] = None,
+        resend_delay: Union[int, None] = None,
+        silent_period: Union[int, None] = None,
+        latest_task_tag: Union[TypedTaskTag, dict[Any, Any], None] = None,
+    ):
+        self.uuid = uuid
+        self.alert_tag_uuid = alert_tag_uuid
+        self.email_address = email_address
+        self.resend_delay = resend_delay
+        self.silent_period = silent_period
+        self.latest_task_tag = latest_task_tag if latest_task_tag is not None else {}
+
+    @classmethod
+    def from_ansible(cls, ansible_data: TypedEmailAlertFromAnsible):
+        return EmailAlert(
+            uuid=ansible_data["uuid"],
+            alert_tag_uuid=ansible_data["alert_tag_uuid"],
+            email_address=ansible_data["email_address"],
+            resend_delay=ansible_data["resend_delay"],
+            silent_period=ansible_data["silent_period"],
+            latest_task_tag=ansible_data["latest_task_tag"],
```

Comment:
I think you do not need to read from ansible fields that are not ansible module parameters.